helm/generic-node/templates/generic-node.yaml

```diff
@@ -0,0 +1,292 @@
+            cpu: {{.Values.resources.requests.cpu}}
+        args:
+        - "mina-daemon"
+        - "run"
+        - "daemon"
+        - "--log-level"
+        - "Debug"
+        - "--log-json"
+        - "--client-port"
+        - "$(DAEMON_CLIENT_PORT)"
+        - "--rest-port"
+        - "$(DAEMON_REST_PORT)"
+        - "--external-port"
+        - "$(DAEMON_EXTERNAL_PORT)"
+        - "--metrics-port"
+        - "$(DAEMON_METRICS_PORT)"
+        - "-run-snark-coordinator"
+        - "$(CODA_SNARK_KEY)"
+        - "-snark-worker-fee"
+        - "$(CODA_SNARK_FEE)"
+        - "-work-selection"
+        - "$(WORK_SELECTION)"
+        - "-u"
+        - "/code/watchdog.py"
+        - "--incluster"
+        - "true"
+        - "--namespace"
+        - ""
+        {{- if .Values.libp2p.staticKeypair.enabled}}
+        - "--discovery-keypair"
+        - "/root/libp2p-keys/key"
+        {{- end }}
+        {{- if .Values.libp2p.seed.enabled}}
+        - "--seed"
+        {{- end }}
+        {{- if .Values.features.runtimeConfig.enabled}}
+        - "--config-file"
+        - "/config/daemon.json"
+        {{- end }}
+        {{- if .Values.features.maxConnections}}
+        - "--max-connections"
+        - {{.Values.features.maxConnections | quote}}
+        {{- end }}
+        {{- range .Values.features.seedPeers}}
+        - "-peer"
+        - {{ . | quote }}
+        {{- end }}
+        {{- if .Values.features.privateKeySecret.enabled}}
+        - "--block-producer-key"
+        - "/root/wallet-keys/key"
+        {{- end }}
+        {{-  if .Values.features.uploadBlocksToGCloud.enabled}}
+        - "-upload-blocks-to-gcloud"
+        - "true"
+        {{- end }}
+        {{-  if .Values.features.persistence.enabled}}
+        - "--config-directory"
+        - "{{.Values.features.persistence.mountPath}}/.mina-config"
+        {{- end }}
+        {{-  if .Values.features.seedPeersLists.enabled}}
+        - "--peer-list-url"
+        - "https://storage.googleapis.com/seed-lists/{{.Values.testnetName}}_seeds.txt"
+        {{- end }}
+        {{- range .Values.daemon.args}}
+        - {{ .| quote }}
+        {{- end }}
+        env:
+        - name: DAEMON_REST_PORT
+          value: {{.Values.ports.graphql | quote}}
+        - name: DAEMON_CLIENT_PORT
+          value: {{.Values.ports.client  | quote}}
+        - name: DAEMON_METRICS_PORT
+          value: {{.Values.ports.metrics | quote}}
+        - name: MINA_CLIENT_TRUSTLIST
+          value: "10.0.0.0/8"
+        - name: CODA_CLIENT_TRUSTLIST
+          value: "10.0.0.0/8"
+        - name: CONNECT_PRIVATE_IPS
+          value: "true"
+        - name: GCLOUD_KEYFILE
+          value: "/gcloud/keyfile.json"
+        - name: NETWORK_NAME
+          value: {{ .Values.testnetName}}
+        - name: GCLOUD_BLOCK_UPLOAD_BUCKET
+          value: "features_network_block_data"
+        - name: DAEMON_EXTERNAL_PORT
+          value: {{.Values.ports.p2p | quote}}
+        - name: CODA_SNARK_KEY
```

Comment:
Hide these (SNARK_KEY, SNARK_FEE) behind the same .Values.feature.snarkCoordinator